Automattic/jetpack
# Social: Fix media validation deadlock

Fixes https://github.com/Automattic/jetpack-reach/issues/478

## Proposed changes:

* Show the post form if
  * There are enabled connections
  * There is any attached media
  * There is any media validation error - to combat the issue of not showing the form with the no media error

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jetpack-reach/issues/478
## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the post editor and check the following scenarios:

Go to an already published post and have only an Instagram connection
* If SIG is off you cannot turn the connection on because it's disabled because of the no media error - deadlock. With this change the media picker should be shown so you can upload media 

Go to an already published post and add a media that is invalid for all connections
* All connections should get disabled, a warning is shown and you are in a deadlock on trunk. With this change the media picker should be visible, so you can remove the image from it

Go to an already published post that has no errors and no media.
* The form should not be shown
